### PR TITLE
Re-implement scons-local build + travis ci updates for build

### PR DESCRIPTION
**Pull request by bdbaddog**
After gutting the SConstruct logic to remove old packaging and move to more modern pip/wheel, it's necessary to reimplement building the scons-local packages.

Also I've added a build job to travis which will build all the packages and docs.

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `CHANGES.txt` (and read the `README.rst`)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -76,4 +76,21 @@ jobs:
         - coverage combine
         - coverage report
         - coverage xml -i -o coverage_xml.xml
-        - codecov -X gcov --file coverage_xml.xml
\ No newline at end of file
+        - codecov -X gcov --file coverage_xml.xml
+
+    - &scons_build_job
+        name: SCons Build
```
Comment (dmoody256) [inline .travis.yml]: this jobs properties are all indented 2 more spaces, could we get them all aligned?

Comment (bdbaddog) [reply .travis.yml]: done

**Review by dirkbaechle — APPROVED** [review]
Looks good, assuming that the new Sphinx stuff simply works (had no time to check this locally, yet).

File: doc/SConscript [hunk 2]
```diff
@@ -480,84 +491,50 @@ else:
...
+    pdf_file = env.Command('#/build/doc/api/latex/SConsAPIDocs.pdf',
+                           env.Glob('#/SCons/*'),
+                           "cd doc && make latexpdf")
+    pdf_install = os.path.join(build, 'PDF', 'scons-api.pdf')
+    env.InstallAs(pdf_install, pdf_file)
+    tar_deps.append(pdf_install)
+    tar_list.append(pdf_install)
+
+    htmldir = os.path.join(build, 'HTML', 'scons-api')
+    html_files = env.Command('#/build/doc/api/html/index.html',
```
Comment (mwichmann) [inline doc/SConscript]: these look reasonable to me. *dirhtml* instead of *html*?

Comment (bdbaddog) [reply doc/SConscript]: dir_html instead of htmldir?

**Comment by bdbaddog** [comment]
O.k. This should be the final form.
To test:

make a virtualenv
pip install -r requirements.txt
python scripts/scons.py
It shouldn't error (at least if you have the proper tetex packages)
It should produce the following in build/dist
```
% ls -l build/dist
total 88952
-rw-r--r--  1 bdbaddog  staff   3816755 Jun 30 16:00 SCons-3.9.9a996-py3-none-any.whl
-rw-r--r--  1 bdbaddog  staff   1912679 Jun 30 16:00 SCons-3.9.9a996.tar.bz2
-rw-r--r--  1 bdbaddog  staff   3842948 Jun 30 16:00 SCons-3.9.9a996.zip
-rw-r--r--  1 bdbaddog  staff  13349459 Jun 30 16:02 scons-doc-3.9.9a996.tar.gz
-rw-r--r--  1 bdbaddog  staff   8926147 Jun 30 16:02 scons-local-3.9.9a996.tar.gz
-rw-r--r--  1 bdbaddog  staff  11911567 Jun 30 16:02 scons-local-3.9.9a996.zip
```

If you change the version in SConstruct (line 40), it should rebuild.
```
default_version = '3.9.9a996'
```
I still have to figure out how to avoid it re-creating SCons/__init__.py when some of the strings passed to TextFile() change.
For now be aware your SCons/__init__.py will change every time you build.

If you can give it a shot, please do.

**Pull request merged**